crypto/rand/rand_unix.c

```diff
@@ -348,50 +381,34 @@ static uint64_t get_time64(void)
         read_wall_time(&t, TIMEBASE_SZ);
         return TWO32TO64(t.tb_high, t.tb_low);
     }
-# elif defined(OSSL_POSIX_TIMER_OKAY)
+# else
+
+#  if defined(OSSL_POSIX_TIMER_OKAY)
     {
         struct timespec ts;
-        clockid_t cid;
 
-#  ifdef CLOCK_BOOTTIME
-        cid = CLOCK_BOOTTIME;
-#  elif defined(_POSIX_MONOTONIC_CLOCK)
-        cid = CLOCK_MONOTONIC;
-#  else
-        cid = CLOCK_REALTIME;
-#  endif
+#   ifdef CLOCK_BOOTTIME
+#    define CLOCK_TYPE CLOCK_BOOTTIME
+#   elif defined(_POSIX_MONOTONIC_CLOCK)
+#    define CLOCK_TYPE CLOCK_MONOTONIC
+#   else
+#    define CLOCK_TYPE CLOCK_REALTIME
+#   endif
 
-        if (clock_gettime(cid, &ts) == 0)
+        if (clock_gettime(CLOCK_TYPE, &ts) == 0)
             return TWO32TO64(ts.tv_sec, ts.tv_nsec);
     }
-# endif
-# if defined(__unix__) \
+#  endif
+#  if defined(__unix__) \
      || (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L)
     {
         struct timeval tv;
 
         if (gettimeofday(&tv, NULL) == 0)
```

Comment:
The manual page says that the call only fails if the args are out of range, which is not the case here. Is it really necessary to check the return value?